lib/kdl/include/kdl/parallel.h

```diff
@@ -61,17 +41,58 @@ namespace kdl {
             threads[i] = std::async(std::launch::async, [&]() {
                 while (true) {
                     const size_t ourIndex = std::atomic_fetch_add(&nextIndex, static_cast<size_t>(1));
-                    if (ourIndex >= input.size()) {
+                    if (ourIndex >= count) {
                         break;
                     }
-                    result[ourIndex] = transform(input[ourIndex]);
+                    lambda(ourIndex);
                 }
             });
         }
 
         for (size_t i = 0; i < numThreads; ++i) {
             threads[i].wait();
         }
+    }
+
+    //template<class T, class L>
```

Comment:
Can this be removed?

Reply:
I think you're looking at a single commit, please check the overall diff

Reply:
But yeah it's removed in a later commit :)